plugins/auth-backend/src/lib/resolvers/CatalogAuthResolverContext.ts

```diff
@@ -123,8 +124,14 @@ export class CatalogAuthResolverContext implements AuthResolverContext {
       const res = await this.catalogApi.getEntities({ filter }, { token });
       result = res.items;
     } else if ('filter' in query) {
+      const filter = lodash.cloneDeep(query.filter);
+      for (const item of [filter].flat()) {
+        if (!('kind' in Object.keys(item).map(lowerCase))) {
```

Comment:
```suggestion
        if (!Object.hasOwn(item, 'kind')) {
```

The nicest way to check this nowadays

Reply:
Please correct me if I'm wrong, but wouldn't this cause issues if someone defined the property as 'Kind' or any other way which is not just lowercased? If I read the documentation of "hasOwn" correctly this is case sensitive and might cause trouble for users defining their own kind in certain situations.